### PR TITLE
Fix RequestHeaderXForwardedForTransform format

### LINKED ISSUE #2347: RequestHeaderXForwardedForTransform wrongly appends existing headers with a ","
The `RequestHeaderXForwardedForTransform` wrongly appends existing headers with a `;` instead of `,`.

```
// Expected
X-Forwarded-For: existing
X-Forwarded-For: existing, proxy

// Actual
X-Forwarded-For: existing 
X-Forwarded-For: existing; proxy
```
Correct format: `X-Forwarded-For: <client>, <proxy1>, <proxy2>`
https://en.wikipedia.org/wiki/X-Forwarded-For

### To Reproduce
Add the following test cases to [RequestHeaderXForwardedForTransformTests](https://github.com/microsoft/reverse-proxy/blob/4397d054801ed7205d3dacc59f5b95b5e0be3caa/test/ReverseProxy.Tests/Transforms/RequestHeaderXForwardedForTransformTests.cs).

```
[InlineData("existing", "127.0.0.1", ForwardedTransformActions.Set, "127.0.0.1")]
[InlineData("existing", "127.0.0.1", ForwardedTransformActions.Append, "existing, 127.0.0.1")]
[InlineData("existing", "127.0.0.1", ForwardedTransformActions.Remove, "")]
```

Also these are wrong:
```
[InlineData("existing,Header", "127.0.0.1", ForwardedTransformActions.Append, "existing,Header;127.0.0.1")]
[InlineData("existing;Header", "127.0.0.1", ForwardedTransformActions.Append, "existing;Header;127.0.0.1")]
```
### Further technical details

- Version: 2.1.0
- Platform: Windows

### PR DESCRIPTION
**Pull request by eddietisma**
Fixes #2347

### CODE REVIEW

**Comment by eddietisma** [comment]
I figured another solution might be:

```diff
private void Append(RequestTransformContext context, string? remoteIp)
{
    var existingValues = TakeHeader(context, HeaderName);
    if (remoteIp is null)
    {
        if (!string.IsNullOrEmpty(existingValues))
        {
            AddHeader(context, HeaderName, existingValues);
        }
    }
    else
    {
-       var values = StringValues.Concat(existingValues, remoteIp);
+       var values = StringValues.Concat(existingValues, remoteIp).ToList();
-       AddHeader(context, HeaderName, values);
+       AddHeader(context, HeaderName, string.Join(", ", values));
    }
}
```

Updated MR.

File: test/ReverseProxy.Tests/Transforms/RequestHeaderXForwardedForTransformTests.cs [hunk 1]
```diff
@@ -13,30 +13,28 @@ namespace Yarp.ReverseProxy.Transforms.Tests;
...
-    [InlineData("existing;Header", "", ForwardedTransformActions.Set, "")]
-    [InlineData("existing,Header", "", ForwardedTransformActions.Append, "existing,Header")]
-    [InlineData("existing;Header", "", ForwardedTransformActions.Append, "existing;Header")]
-    [InlineData("existing;Header", "", ForwardedTransformActions.Remove, "")]
-    [InlineData("existing,Header", "127.0.0.1", ForwardedTransformActions.Set, "127.0.0.1")]
-    [InlineData("existing;Header", "127.0.0.1", ForwardedTransformActions.Set, "127.0.0.1")]
-    [InlineData("existing,Header", "127.0.0.1", ForwardedTransformActions.Append, "existing,Header;127.0.0.1")]
-    [InlineData("existing;Header", "127.0.0.1", ForwardedTransformActions.Append, "existing;Header;127.0.0.1")]
-    [InlineData("existing;Header", "127.0.0.1", ForwardedTransformActions.Remove, "")]
+    [InlineData("existing, Header", "", ForwardedTransformActions.Set, "")]
```
Comment (Tratcher) [inline test/ReverseProxy.Tests/Transforms/RequestHeaderXForwardedForTransformTests.cs:23]: The ';' was a test artifact, not something that would have shown up in the actual output.

**Review by Tratcher — CHANGES_REQUESTED** [review]
Did you see this happen in a real app or were you just looking at the tests? I think you mis-understood the test code, it was just using ';' to serialize test data, production data doesn't do that.

**Comment by eddietisma** [comment]
Yeah, a big misunderstanding from my part. It is working as expected. Sorry!

**Pull request closed without merging**